### PR TITLE
Update ember-collapsible-panel for ember >= 3.3.0

### LINKED ISSUE #47: Uncaught Error: unreachable
Not really sure what's going on, but in a fresh ember app, I'm getting the following error:
image

my project is simply:
```bash
$ ember -v
ember-cli: 3.3.0
node: 8.11.3
os: darwin x64

$ ember new scratch
...

$ cd scratch
$ ember install ember-collapsible-panel
...
[replace application.hbs with "basic usage" example from the docs site]
```

I can give you a repo if you like. I have a hunch it has to do with the module unification/pod component structure.

### PR DESCRIPTION
**Pull request by iand675**
Fixes #47 

This PR updates the ember-collapsible-panel documentation in order to work properly in light of this issue: https://github.com/emberjs/ember.js/issues/16826

In short, `p` as an argument to blocks no longer works, so all usages have been update to use `panel`. This should help new users not be confused about why attempting to copy and paste from the docs doesn't work.

### CODE REVIEW

File: testem.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   test_page: 'tests/index.html?hidepassed',
+  client_decycle_depth: 12,
```
Comment (Alonski) [inline testem.js:3]: Hmm why is this needed?

**Comment by Alonski** [comment]
@iand675 Maybe we can separate the doc changes from the update changes?
They aren't dependent on each other.

**Comment by iand675** [comment]
@Alonski I don't understand why you'd want to? If the docs are wrong, then people can't effectively use the addon. In any case, they're separate commits so they'd show up separately when you merge the PR.

File: testem.js (hunk 1 above)
Comment (iand675) [reply testem.js:3]: Ask me half a year ago :smile:. I'm afraid I don't remember at this point, but I'd expect it had to do with the default `client_decycle_depth` changing when bumping deps and thus causing test failures.

**Comment by Alonski** [comment]
@samselikoff What do you say about getting this merged in?
I don't think the failing tests make sense. Can you rerun Travis?

**Comment by samselikoff** [comment]
@Alonski just kicked off the build again.

I'm realizing that we don't really use this addon, we inherited from when we built it at TED several years ago but at this point we're not really in a spot to be maintaining it. Would you have an interest in taking over maintenance?

If not we can start putting out a wider call to look for maintainers.

**Comment by Alonski** [comment]
@samselikoff I'd love to take it over 😁 I think I offered to a few months ago.

I use it heavily in my apps.

**Comment by samselikoff** [comment]
Yes I remember! Sorry I have the short term memory of a fish. 

I think at the time we had plans to do more maintenance but we’re trying to eliminate some things and refocus on others. Will get back to you soon.

**Comment by Alonski** [comment]
@iand675 I have taken over maintaining the addon. Sorry but I made a few merge conflicts :)
Can you please merge/rebase and push again?

**Comment by Alonski** [comment]
Closed via #60 
Thanks @iand675 for the initial work!
